modules/sso.php

```diff
@@ -554,6 +554,44 @@ static function clear_wpcom_profile_cookies() {
 		}
 	}
 
+	/**
+	 * Clear cookies that are no longer needed once the user has logged in.
+	 *
+	 * @since 4.8.0
+	 */
+	static function clear_cookies_after_login() {
+		self::clear_wpcom_profile_cookies();
+		if ( isset( $_COOKIE[ 'jetpack_sso_nonce' ] ) ) {
+			setcookie(
```

Comment:
The cookie is deleted immediately after the user successfully logs in. And if the user doesn't log in, the cookie is only good for 10 minutes.

Further, the nonce isn't tied to any user. It's used to verify intent. So, I think it's OK to store in it in the cookie temporarily.